adapters/risemediatech/risemediatech.go

```diff
@@ -0,0 +1,162 @@
+package risemediatech
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/iterutil"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, cfg config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{endpoint: cfg.Endpoint}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+	var setTestMode bool
+
+	for imp := range iterutil.SlicePointerValues(request.Imp) {
+		impExt, err := parseImpExt(imp.Ext)
+		if err != nil {
+			errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: %v", imp.ID, err)})
+			continue
+		}
+
+		if imp.Banner != nil {
+			if imp.Banner.W == nil || imp.Banner.H == nil || *imp.Banner.W == 0 || *imp.Banner.H == 0 {
+				errs = append(errs, &errortypes.BadInput{Message: fmt.Sprintf("impID %s: invalid banner dimensions", imp.ID)})
+				continue
+			}
+		}
```

Comment:
There are banner validations that take place upstream and if they result in an error your adapter will not be called. Please see if these suffice, in which case you should remove this logic:
https://github.com/prebid/prebid-server/blob/master/ortb/request_validator_banner.go